explainability-core/src/main/java/org/kie/trustyai/explainability/local/tssaliency/TSSaliencyExplainer.java

```diff
@@ -192,51 +196,51 @@ public RealMatrix monteCarloGradient(RealMatrix x, PredictionProvider model) thr
 
         final int T = x.getRowDimension();
         final int F = x.getColumnDimension();
-
-        final NormalDistribution N = new NormalDistribution(randomGenerator, 0.0, sigma);
-
-        final RealMatrix[] U = new RealMatrix[ng];
-        for (int i = 0; i < ng; i++) {
-            U[i] = MatrixUtils.createRealMatrix(T, F); // Creates a matrix with T rows and F columns
-        }
+        final Random r = new Random();
```

Comment:
I remember we used to hide/centralise random generation inside [`PerturbationContext`](https://github.com/trustyai-explainability/trustyai-explainability/blob/main/explainability-core/src/main/java/org/kie/trustyai/explainability/model/PerturbationContext.java#L55). 
Do you think we can adopt a similar mechanism also here?